WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/Create Shipping Label/ShippingLabelAddNewPackageViewModelTests.swift

```diff
@@ -0,0 +1,89 @@
+import XCTest
+@testable import Yosemite
+@testable import WooCommerce
+
+class ShippingLabelAddNewPackageViewModelTests: XCTestCase {
+
+    private let sampleSiteID = 12345
+
+    func test_createCustomPackage_resets_child_view_models_on_success() {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let viewModel = ShippingLabelAddNewPackageViewModel(stores: stores,
+                                                            siteID: 12345,
+                                                            packagesResponse: ShippingLabelPackagesResponse.fake(),
+                                                            onCompletion: {_, _, _ in })
+
+        // Given a validated custom package
+        viewModel.customPackageVM.packageName = "Test Package"
+        viewModel.customPackageVM.packageHeight = "1"
+        viewModel.customPackageVM.packageWidth = "1"
+        viewModel.customPackageVM.packageLength = "1"
+        viewModel.customPackageVM.emptyPackageWeight = "1"
+
+        // When
+        stores.whenReceivingAction(ofType: ShippingLabelAction.self) { action in
+            switch action {
+            case let .createPackage(_, _, _, onCompletion):
+                onCompletion(.success(true))
+            case let .packagesDetails(_, onCompletion):
+                onCompletion(.success(ShippingLabelPackagesResponse.fake()))
+            default:
+                break
+            }
+        }
+
+        viewModel.createCustomPackage(onCompletion: { _ in })
+
+        // Then
+        XCTAssertNil(viewModel.customPackageVM.validatedCustomPackage)
```

Comment:
Suggestion: it'd be nice to also assert that all fields are reset to default values?